py/tools/pex/BUILD.bazel

```diff
@@ -0,0 +1,9 @@
+load("@rules_python//python:defs.bzl", "py_binary")
+
+py_binary(
+    name = "pex",
+    srcs = ["main.py"],
+    main = "main.py",
+    deps = ["@pypi_pex//:pkg"],
```

Comment:
Where does the user get this dependency from? 

rules_py currently doesn't have any Python dependencies, this will be the first, so there is no call to a pip install in the setup. This is working here as it's using the internal test time dependencies that rules_py uses.

Ideally we wouldn't have a Python dependency at all, is there another way to interact with the PEX library, can we vendor the dependency?

Reply:
You are right, my latest commit fixes this by introducing a slightly complicated bootstrapping logic to allow pex builder to be either a py_binary from rules_python with a pip dependency (for local dev only) or a py_binary from rules_python with no external dependencies but a  pex builder packaged into a pex zip to prevent a python dependency.

Reply:
> Ideally we wouldn't have a Python dependency at all, is there another way to interact with the PEX library, can we vendor the dependency?

vendoring is another approach we could take, it's open for a debate.

Reply:
Given that pex is a single wheel with no dependencies, I'm not clear why we need to bootstrap it via a zip? 
Could we instead fetch the wheel via an `htt_file`, have rules_py expand it with the existing tools, then depend on that? rules_py has a fetch dependencies function where it declares other rule dependencies, so we could add the `http_file` into there.

Reply:
Yeah, that sounds better, i'll do that once get back to this again.

Reply:
This is now fixed.